lib/page_ez/visitors/registered_name_visitor.rb

```diff
@@ -17,16 +19,21 @@ def define_method(*)
       def inherit_from(*)
       end
 
-      def process_macro(_, name, _)
-        if @declared_names.include?(name)
+      def process_macro(_, name, construction_strategy)
+        @declared_constructors[@parents.last.object_id] ||= []
+
+        if @declared_constructors[@parents.last.object_id].include?(name)
```

Comment:
The tests continue to pass with this change and I think it's a bit easier to think about?

```suggestion
      def process_macro(_, name, construction_strategy)
        @declared_constructors[@parents.last] ||= []

        if @declared_constructors[@parents.last].include?(name)
```